src/routes/docs/tutorials/pagination/step-8/+page.markdoc

```diff
@@ -0,0 +1,275 @@
+        setLastId(response.documents[response.documents.length - 1].$id);
+        setTodos([...todos, ...response.documents]);
+    } catch (error) {
+      console.log("Error loading more todos:", error);
+    }
+    }
+  }
+  async function loadPrev() {
+  if (currentPage > 1) {
+    try {
+        const response = await databases.listDocuments(
+          TODOS_DATABASE_ID,
+          TODOS_COLLECTION_ID,
+          [
+            Query.orderAsc("$createdAt"),
+            Query.limit(pageLimit),
+            Query.cursorBefore(firstId),
+          ],
+        );
+        setCurrentPage(currentPage -1 );
+        setFirstId(response.documents[0].$id);
+        setLastId(response.documents[response.documents.length - 1].$id);
+        setTasks(response.documents);
+    } catch (error) {
+      console.error("Error loading previous todos:", error);
+    }
+  }
+  }
+  async function init() {
+    try {
+      const response = await databases.listDocuments(
+        TODOS_DATABASE_ID,
+        TODOS_COLLECTION_ID,
+        [Query.orderAsc("$createdAt"), Query.limit(pageLimit)],
+      );
+      setFirstId(response.documents[0].$id);
+      setLastId(response.documents[response.documents.length - 1].$id);
+      setTasks(response.documents);
+      setTotalPages(Math.ceil(response.total / pageLimit));
+    } catch (error) {
+      console.log("Error initializing tasks:", error);
+    }
+  }
+  useEffect(() => {
+    init();
+  }, []);
+
+  return (
+    <TodosContext.Provider
+      value={{
+        current: todos,
+        loadMore,
+        loadPrev,
+        currentPage,
+        totalPages,
+        hasNext,
+      }}>
+      {props.children}
+    </TodosContext.Provider>
+  );
+}
+```
+In this updated context:
+ * We're using `Query.cursorAfter()` is used to fetch documents that come after a specific document.
+ * We're using `Query.cursorBefore()` is used to fetch documents that come before a specific document.
+ * We've added `lastId` to hold the ID of the last document in the current set of todos.By using this ID with `Query.cursorAfter()`, We're essentially telling Appwrite: "Give me the next set of todos that come after this specific todo item."
+ * We's added `firstId` to hold the ID of the first document in the current set of todos. By using this ID with `Query.cursorBefore()`, we're telling Appwrite: "Give me the set of todos that come before this specific todo item."
+
+Now, you can use this context in your components to display todo items and handle pagination using Appwrite's cursor pagination methods.
+# Update Todos page {% #update-todos-page %}
+Now, let's use our Pagination component in the Todos Page.
+{% tabs %}
+{% tabsitem #Basic-cursor-pagination title="Basic Cursor Pagination" %}
+Update the Todos page in `src/pages/Todos.jsx`:
+```js
+import React from 'react';
+import { useTodos } from '../lib/context/todos';
+import BasicCursorPagination from '../components/BasicCursorPagination';
+
+export function TodosPage() {
```

Comment:
The name again should be `Todos` cause that's what App.jsx imports

Reply:
```suggestion
export function Todos() {
```